### PR TITLE
Droth-659 point like asset to complementary link

### PR DESCRIPTION
**Pull request by MAC-CGI**
Droth-659 point like asset to complementary link

### CODE REVIEW

**Comment by halied** [comment]
Ok

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RoadLinkService.scala [hunk 1]
```diff
@@ -1210,6 +1247,37 @@ class RoadLinkService(val vvhClient: VVHClient, val eventbus: DigiroadEventBus,
     }
   }
 
+  private def getCachedRoadLinksWithComplementaryAndChanges(municipalityCode: Int): (Seq[RoadLink], Seq[ChangeInfo]) = {
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RoadLinkService.scala]: Duplicated code for the cache.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RoadLinkService.scala [hunk 2]
```diff
@@ -1210,6 +1247,37 @@ class RoadLinkService(val vvhClient: VVHClient, val eventbus: DigiroadEventBus,
...
+        val (roadLinks, changes) = reloadRoadLinksWithComplementaryAndChangesFromVVH(municipalityCode)
+        if (dir.nonEmpty) {
+          try {
+            val newGeomFile = new File(dir.get, geometryWithComplementaryCacheFileNames.format(municipalityCode, System.currentTimeMillis))
+            if (vvhSerializer.writeCache(newGeomFile, roadLinks)) {
+              logger.info("New cached file created: " + newGeomFile + " containing " + roadLinks.size + " items")
+            } else {
+              logger.error("Writing cached geom file failed!")
+            }
+            val newChangeFile = new File(dir.get, changeCacheFileNamesWithComplementary.format(municipalityCode, System.currentTimeMillis))
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RoadLinkService.scala]: The change info with or without complementary is the same, so we don't need to have a separated cache for it.

**Comment by joanacoelho** [comment]
ok

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/pointasset/oracle/OracleObstacleDao.scala [hunk 3]
```diff
@@ -40,7 +43,7 @@ object OracleObstacleDao {
 
   implicit val getPointAsset = new GetResult[Obstacle] {
     def apply(r: PositionedResult) = {
-      val id = r.nextLong()
+      val id = r.nextLong
```
Comment (marquesfm) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/pointasset/oracle/OracleObstacleDao.scala]: why we  remove the "()" from the nextLong()?

Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/pointasset/oracle/OracleObstacleDao.scala]: Because you don't need it

**Comment by MAC-CGI** [comment]
Pull request close because it was created a new branch with all VVHclient changes

**Pull request closed without merging**